src/hooks/useInflation.ts

```diff
@@ -1,22 +1,106 @@
   };
 
-  return { activeInflationConfiguration, fetchActiveConfigurationToStore };
+  /**
+   * Estimates the realized inflation rate percentage based on the actual total issuance at the beginning
+   * and estimated total issuance at the end of the current cycle.
+   * According to the https://github.com/AstarNetwork/astar-apps/issues/1259
+   */
+  const estimateRealizedInflation = async (): Promise<number | undefined> => {
+    let inflation: number | undefined;
+
+    if ($api) {
+      try {
+        // Find the block when last NewInflationConfiguration event was emitted.
+        const subscanRepository = container.get<ISubscanRepository>(Symbols.SubscanRepository);
+        const response = await subscanRepository.getEvents(
+          networkNameSubstrate.value.toLocaleLowerCase(),
+          'inflation',
+          'NewInflationConfiguration'
+        );
+        // Latest item in array is for the current inflation cycle.
+        const latestInflationConfigBlock = response.events[response.events.length - 1].block;
+
+        const initialIssuanceBlockHash = await $api.rpc.chain.getBlockHash(
+          latestInflationConfigBlock - 1
+        );
+        const apiAt = await $api.at(initialIssuanceBlockHash);
+        const initialTotalIssuance = await apiAt.query.balances.totalIssuance();
+        const realizedTotalIssuance = await $api.query.balances.totalIssuance();
+        const {
+          periodsPerCycle,
+          standardEraLength,
+          standardErasPerBuildAndEarnPeriod,
+          standardErasPerVotingPeriod,
+        } = eraLengths.value;
+        const cycleLengthInBlocks =
+          standardEraLength *
+          periodsPerCycle *
+          (standardErasPerBuildAndEarnPeriod + standardErasPerVotingPeriod);
+        const blockDifference = BigInt(currentBlock.value - latestInflationConfigBlock);
+        const slope =
+          BigInt(realizedTotalIssuance.sub(initialTotalIssuance).toString()) / blockDifference;
+
+        // Estimate total issuance at the end of the current cycle.
+        const endOfCycleBlock = latestInflationConfigBlock + cycleLengthInBlocks;
+        const endOfCycleTotalIssuance = Number(
+          ethers.utils.formatEther(
+            slope * BigInt(endOfCycleBlock - latestInflationConfigBlock) +
+              initialTotalIssuance.toBigInt()
+          )
+        );
+
+        // Estimated inflation at the end of the current cycle.
+        inflation =
+          (100 *
+            (endOfCycleTotalIssuance -
+              Number(ethers.utils.formatEther(initialTotalIssuance.toString())))) /
+          endOfCycleTotalIssuance;
+      } catch (error) {
+        console.error('Error calculating realized inflation', error);
+      }
+    }
+
+    return inflation;
+  };
+
+  watch(
+    [$api],
+    async () => {
+      estimatedInflation.value = await estimateRealizedInflation();
+      console.log('Realized inflation:', estimatedInflation.value);
```

Comment:
Let's use `console.info` if you want to show this in the console

Reply:
Actually, this is not needed. Removed in https://github.com/AstarNetwork/astar-apps/pull/1280/commits/e120bdb75d843a397f4821a7f6ebf1ba47b229e8